packages/data/lib/dio/token_interceptor.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:dio/dio.dart';
+import 'package:domain/repository/local_base_repo.dart';
+
+class TokenInterceptor extends Interceptor {
+  final ILocalStorageRepo _localStorageRepository;
+
+  TokenInterceptor(this._localStorageRepository);
+
+  @override
+  void onRequest(
+    RequestOptions options,
+    RequestInterceptorHandler handler,
+  ) async {
+    options.headers['Authorization'] = await _localStorageRepository.getToken();
+    return super.onRequest(options, handler);
+  }
+
+  @override
+  void onError(
+    DioError err,
+    ErrorInterceptorHandler handler,
+  ) async {
+    if (err.response?.statusCode == 401) {
+      await _localStorageRepository.deleteToken();
+    }
+    return super.onError(err, handler);
+  }
+
+  @override
+  void onResponse(Response response, ResponseInterceptorHandler handler) {
+    //TODO impl response
+
+    super.onResponse(response, handler);
+  }
+}
```

Comment:
why do u need to override onResponse?

Reply:
onResponse I am trying impl that case what you told me, what if user open the app first time and leave from the login page without log in

I want save the token onResponse if is 200
if it is 401 delete

or is it not good idea?